metrics/ci_integrations/test/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate_test.dart

```diff
@@ -0,0 +1,360 @@
+import 'package:ci_integration/destination/firestore/strings/firestore_strings.dart';
+import 'package:firedart/firedart.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_utils/matchers.dart';
+import '../../test_utils/mock/firebase_auth_mock.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("FirestoreDestinationValidationDelegate", () {
+    const apiKey = 'key';
+    const email = 'stub@email.com';
+    const password = 'stub_password';
+    const message = 'message';
+
+    final credentials = FirebaseAuthCredentials(
+      apiKey: apiKey,
+      email: email,
+      password: password,
+    );
+
+    const invalidApiKeyAuthException = FirebaseAuthException(
+      FirebaseAuthExceptionCode.invalidApiKey,
+      message,
+    );
+    const emailNotFoundAuthException = FirebaseAuthException(
+      FirebaseAuthExceptionCode.emailNotFound,
+      message,
+    );
+    const invalidPasswordAuthException = FirebaseAuthException(
+      FirebaseAuthExceptionCode.invalidPassword,
+      message,
+    );
+    const passwordLoginDisabledAuthException = FirebaseAuthException(
+      FirebaseAuthExceptionCode.passwordLoginDisabled,
+      message,
+    );
+    const userDisabledAuthException = FirebaseAuthException(
+      FirebaseAuthExceptionCode.userDisabled,
+      message,
+    );
+
+    const invalidAuthExceptionCodes = [
+      FirebaseAuthExceptionCode.emailNotFound,
+      FirebaseAuthExceptionCode.invalidPassword,
+      FirebaseAuthExceptionCode.passwordLoginDisabled,
+      FirebaseAuthExceptionCode.userDisabled,
+    ];
+
+    final user = User.fromMap(const {});
+
+    final authFactory = _FirebaseAuthFactoryMock();
+    final firebaseAuth = FirebaseAuthMock();
+
+    final delegate = FirestoreDestinationValidationDelegate(authFactory);
+
+    PostExpectation<FirebaseAuth> whenCreateFirebaseAuth() {
+      return when(authFactory.create(apiKey));
+    }
+
+    PostExpectation<Future<User>> whenSignIn() {
+      whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+      return when(firebaseAuth.signIn(email, password));
+    }
+
+    tearDown(() {
+      reset(authFactory);
+      reset(firebaseAuth);
+    });
+
+    test(
+      "throws an ArgumentError if the given auth factory is null",
+      () {
+        expect(
+          () => FirestoreDestinationValidationDelegate(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given parameters",
+      () {
+        final validationDelegate = FirestoreDestinationValidationDelegate(
+          authFactory,
+        );
+
+        expect(validationDelegate.authFactory, equals(authFactory));
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() creates a firebase auth with the firebase auth factory",
```

Comment:
```suggestion
      ".validatePublicApiKey() creates a firebase auth using the firebase auth factory",
```
Please, apply this suggestion to all similar situations.